### PR TITLE
Add `dataset.cluster()`

### PR DESCRIPTION
**Pull request by dsmilkov**
- Add `dataset.cluster()`, no UI yet
- Fix a bug with signal menu in schema
- Fix a bug with pagination of groups
- Make the group by header visually aligned with the row header
- Add `Signal.output_type` which can be embedding or a cluster

### CODE REVIEW

File: lilac/data/dataset.py [hunk 1]
```diff
@@ -454,6 +455,11 @@ def compute_signal(
     """
     pass
 
+  def cluster(self, embedding: str, path: Path) -> None:
```
Comment (nsthorat) [inline lilac/data/dataset.py:458]: maybe path first

and maybe we should add a type here with embedding optional? minhash is pretty popular

File: web/blueprint/src/lib/components/contextMenu/SchemaFieldMenu.svelte [hunk 2]
```diff
@@ -79,7 +81,7 @@
           })}
       />
     {/if}
-    {#if !isSignal}
+    {#if isComputable}
```
Comment (nsthorat) [inline web/blueprint/src/lib/components/contextMenu/SchemaFieldMenu.svelte:84]: maybe canComputeSignal?

**Review by nsthorat — APPROVED** [review]

File: lilac/data/dataset.py (hunk 1 above)
Comment (dsmilkov) [reply lilac/data/dataset.py]: done

File: web/blueprint/src/lib/components/contextMenu/SchemaFieldMenu.svelte (hunk 2 above)
Comment (dsmilkov) [reply web/blueprint/src/lib/components/contextMenu/SchemaFieldMenu.svelte:84]: done

**Pull request merged**